router.go

```diff
@@ -1 +1,240 @@
 package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+)
+
+type MyMux struct {
+}
+
+func (p *MyMux) ServeHTTP(r *Request) string {
+	switch r.URL.Path {
+	case "/head_request":
```

Comment:
i would recommend refactoring here.. a 'router' class should only be routing the request based on endpoint and request method to a handler. you should delegate handling the response (building the response status, headers, body, etc) to a diff file.